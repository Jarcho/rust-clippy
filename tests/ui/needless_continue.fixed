//@aux-build:proc_macros.rs

#![warn(clippy::needless_continue)]
#![allow(while_true, clippy::empty_loop)]
use proc_macros::{external, inline_macros, with_span};

#[inline_macros]
fn main() {
    loop { //~ needless_continue
    }

    while true {
        let _ = 1 + 1; //~ needless_continue
    }

    for _ in 0..1 {
        let _ = match 0 {
            0 => 0,
            1 => 1,
            _ => 2,
        }; //~ needless_continue
    }

    loop {
        if true { //~ needless_continue
        }
    }

    loop {
        if true { //~ needless_continue
        } else {
            println!();
        }
    }

    loop {
        if true {
            println!();
        } else { //~ needless_continue
        }
    }

    loop {
        match 0 {
            0 => break,
            1 => {}, //~ needless_continue
            _ => {},
        }
    }

    loop {
        if true {
            continue;
        }
        match 0 {
            0 => match 0 {
                0 => {}, //~ needless_continue
                1 => {
                    if true { //~ needless_continue
                    } else {
                        break;
                    }
                },
                _ => {},
            },
            1 => {
                let _ = 1; //~ needless_continue
            },
            _ => break,
        }
    }

    'outer: loop {
        'inner: loop {
            match 0 {
                0 => continue 'outer,
                1 => {}, //~ needless_continue
                _ => {},        //~ needless_continue
            }
        }
    }

    'outer: loop {
        loop {
            match 0 {
                0 => continue 'outer,
                1 => {}, //~ needless_continue
                _ => break,
            }
        }
    }

    external! {
        loop {
            continue;
        }
    }

    with_span! {
        span
        loop {
            continue;
        }
    }

    inline! {
        loop { //~ needless_continue
        }

        loop $(@block {
            continue;
        })

        loop {
            $(@stmt continue)
        }

        loop {
            $(@expr continue);
        }

        loop {
            if true $(@block {
                continue;
            })
        }

        loop {
            if true {
                $(@stmt continue)
            } else { //~ needless_continue
            }
        }

        loop {
            $(@stmt match 0 {
                0 => continue,
                1 => break,
                _ => {},
            })
        }
    }
}
