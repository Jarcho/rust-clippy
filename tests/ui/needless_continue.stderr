error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:10:9
   |
LL |       loop {
   |  ___________-
LL | |         continue;
   | |_________^^^^^^^^- help: remove this
   |
   = note: `-D clippy::needless-continue` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::needless_continue)]`

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:15:9
   |
LL |           let _ = 1 + 1;
   |  _______________________-
LL | |         continue;
   | |_________^^^^^^^^- help: remove this

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:24:9
   |
LL |           };
   |  ___________-
LL | |         continue;
   | |_________^^^^^^^^- help: remove this

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:29:13
   |
LL |           if true {
   |  __________________-
LL | |             continue;
   | |_____________^^^^^^^^- help: remove this

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:35:13
   |
LL |           if true {
   |  __________________-
LL | |             continue;
   | |_____________^^^^^^^^- help: remove this

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:45:13
   |
LL |           } else {
   |  _________________-
LL | |             continue;
   | |_____________^^^^^^^^- help: remove this

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:52:18
   |
LL |             1 => continue,
   |                 -^^^^^^^^
   |                 |
   |                 help: remove this: `{}`

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:63:22
   |
LL |                 0 => continue,
   |                     -^^^^^^^^
   |                     |
   |                     help: remove this: `{}`

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:66:25
   |
LL |                       if true {
   |  ______________________________-
LL | |                         continue;
   | |_________________________^^^^^^^^- help: remove this

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:75:17
   |
LL |                   let _ = 1;
   |  ___________________________-
LL | |                 continue;
   | |_________________^^^^^^^^- help: remove this

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:85:22
   |
LL |                 1 => continue 'inner,
   |                     -^^^^^^^^^^^^^^^
   |                     |
   |                     help: remove this: `{}`

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:86:22
   |
LL |                 _ => continue,
   |                     -^^^^^^^^
   |                     |
   |                     help: remove this: `{}`

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:95:22
   |
LL |                 1 => continue,
   |                     -^^^^^^^^
   |                     |
   |                     help: remove this: `{}`

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:116:13
   |
LL |           loop {
   |  _______________-
LL | |             continue;
   | |_____________^^^^^^^^- help: remove this
   |
   = note: this error originates in the macro `__inline_mac_fn_main` (in Nightly builds, run with -Z macro-backtrace for more info)

error: this `continue` expression is redundant
  --> tests/ui/needless_continue.rs:141:17
   |
LL |               } else {
   |  _____________________-
LL | |                 continue;
   | |_________________^^^^^^^^- help: remove this
   |
   = note: this error originates in the macro `__inline_mac_fn_main` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 15 previous errors

